master/buildbot/db/dbconfig.py

```diff
@@ -0,0 +1,87 @@
+# This file is part of Buildbot.  Buildbot is free software: you can
+# redistribute it and/or modify it under the terms of the GNU General Public
+# License as published by the Free Software Foundation, version 2.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, write to the Free Software Foundation, Inc., 51
+# Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Copyright Buildbot Team Members
+
+from buildbot.db import enginestrategy
+from buildbot.db import model
+from buildbot.db import state
+from sqlalchemy.exc import OperationalError
+
+
+class FakeDBConnector(object):
+    pass
+
+
+class FakeCacheManager(object):
+
+    def get_cache(self, cache_name, miss_fn):
+        return None
+
+
+class FakeMaster(object):
+    pass
+
+
+class FakePool(object):
+    pass
+
+
+class DbConfig(object):
+
+    def __init__(self, BuildmasterConfig, basedir, name="config"):
+        if 'db' in BuildmasterConfig:
```

Comment:
my concern is that if we decide to change available options, these will have to be changed in two places.